pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -367,6 +371,8 @@ pub mod pallet {
 			spender: T::AccountId,
 			amount: U256,
 		) -> Result<(), evm::EvmError> {
+			// We perform the evm call in a storage transaction to ensure that if it fail
+			// any contract storage changes are rolled back.
 			EvmCaller::<T>::erc20_approve(
```

Comment:
Not related to this PR, but I noticed that this call is not using `frame_support::storage::with_storage_layer`